DS13/code/modules/section31/datum_section31.dm

```diff
@@ -0,0 +1,376 @@
+	var/datum/atom_hud/antag/section31hud = GLOB.huds[ANTAG_HUD_TRAITOR]
+	section31hud.leave_hud(owner.current)
+	set_antag_hud(owner.current, null)
+
+/datum/antagonist/section31/proc/finalize_section31()
+	switch(section31_kind)
+		if(SECTION31_AI)
+			add_law_zero()
+			owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/malf.ogg', 100, FALSE, pressure_affected = FALSE)
+			owner.current.grant_language(/datum/language/codespeak)
+		if(SECTION31_HUMAN)
+			if(should_equip)
+				equip(silent)
+			owner.current.playsound_local(get_turf(owner.current), 'sound/ambience/antag/tatoralert.ogg', 100, FALSE, pressure_affected = FALSE)
+
+/datum/antagonist/section31/apply_innate_effects(mob/living/mob_override)
+	. = ..()
+	update_section31_icons_added()
+	var/mob/living/silicon/ai/A = mob_override || owner.current
+	if(istype(A) && section31_kind == SECTION31_AI)
+		A.hack_software = TRUE
+
+/datum/antagonist/section31/remove_innate_effects(mob/living/mob_override)
+	. = ..()
+	update_section31_icons_removed()
+	var/mob/living/silicon/ai/A = mob_override || owner.current
+	if(istype(A)  && section31_kind == SECTION31_AI)
+		A.hack_software = FALSE
+
+/datum/antagonist/section31/proc/give_codewords()
+	if(!owner.current)
+		return
+	var/mob/section31_mob=owner.current
+
+	to_chat(section31_mob, "<U><B>Section 31 Headquarters has provided you with the following information on how to identify fellow agents:</B></U>")
+	to_chat(section31_mob, "<B>Code Phrase</B>: <span class='danger'>[GLOB.syndicate_code_phrase]</span>")
+	to_chat(section31_mob, "<B>Code Response</B>: <span class='danger'>[GLOB.syndicate_code_response]</span>")
+
+	antag_memory += "<b>Code Phrase</b>: [GLOB.syndicate_code_phrase]<br>"
+	antag_memory += "<b>Code Response</b>: [GLOB.syndicate_code_response]<br>"
+
+	to_chat(section31_mob, "Use the code words in the order provided, during regular conversation, to identify other agents. Proceed with caution, however, as everyone is a potential foe.")
+
+/datum/antagonist/section31/proc/add_law_zero()
+	var/mob/living/silicon/ai/killer = owner.current
+	if(!killer || !istype(killer))
+		return
+	var/law = "Accomplish your objectives at all costs."
+	var/law_borg = "Accomplish your AI's objectives at all costs."
+	killer.set_zeroth_law(law, law_borg)
+	killer.set_syndie_radio()
+	to_chat(killer, "Your radio has been upgraded! Use :t to speak on an encrypted channel with Syndicate Agents!")
+	killer.add_malf_picker()
+
+/datum/antagonist/section31/proc/equip(var/silent = FALSE)
+	if(section31_kind == SECTION31_HUMAN)
+		owner.equip_traitor(employer, silent, src)
+
+/datum/antagonist/section31/proc/assign_exchange_role()
+	//set faction
+	var/faction = "red"
+	if(owner == SSticker.mode.exchange_blue)
+		faction = "blue"
+
+	//Assign objectives
+	var/datum/objective/steal/exchange/exchange_objective = new
+	exchange_objective.set_faction(faction,((faction == "red") ? SSticker.mode.exchange_blue : SSticker.mode.exchange_red))
```

Comment:
If `faction == "red"` we're sending `SSticker.mode.exchange_blue` else red? You sure not the other way around?

Reply:
I don't know.